mobile_verifier/src/owner_shares.rs

```diff
@@ -14,6 +15,80 @@ use rust_decimal::prelude::*;
 use std::collections::HashMap;
 use std::ops::Range;
 
+pub struct TransferRewards {
+    scale: Decimal,
+    rewards: HashMap<PublicKeyBinary, Decimal>,
+    pub remaining_rewards: Decimal,
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum TransferRewardsError {
+    #[error("Clock error: {0}")]
+    ClockError(#[from] ClockError),
+    #[error("Decimal error: {0}")]
+    DecimalError(#[from] rust_decimal::Error),
+}
+
+impl TransferRewards {
+    pub fn empty(epoch: &Range<DateTime<Utc>>) -> Self {
+        Self {
+            scale: Decimal::ONE,
+            rewards: HashMap::new(),
+            remaining_rewards: get_scheduled_tokens(epoch.start, epoch.end - epoch.start).unwrap(),
+        }
+    }
+
+    pub fn reward(&self, hotspot: &PublicKeyBinary) -> Decimal {
+        self.rewards.get(hotspot).copied().unwrap_or(Decimal::ZERO) * self.scale
+    }
+
+    pub async fn from_transfer_sessions(
+        dollars_per_mobile: u64,
+        transfer_sessions: impl Stream<Item = ValidDataTransferSession>,
+        epoch: &Range<DateTime<Utc>>,
+    ) -> Result<Self, TransferRewardsError> {
+        tokio::pin!(transfer_sessions);
+
+        let mut bytes = HashMap::<PublicKeyBinary, Decimal>::new();
```

Comment:
I have flipped to marc's side and I think it would be more readable as 
```rust
        let bytes: HashMap<PublicKeyBinary, Decimal> = transfer_sessions
            .fold(HashMap::new(), |mut acc, next| async move {
                *acc.entry(next.pub_key).or_default() +=
                    Decimal::from(next.upload_bytes + next.download_bytes);
                acc
            })
            .await;
```